packages/driver/src/dom/visibility.js

```diff
@@ -185,19 +195,28 @@ const elIsHiddenByTransform = ($el) => {
   if (transform.startsWith('matrix3d')) {
     const m3d = transform.substring(8).match(numberRegex)
 
-    // Z Axis values
-    if (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0) {
+    // Test scale to 0
+    if ((+m3d[0] === 0 && +m3d[4] === 0 && +m3d[8] === 0) || // X Axis
+    (+m3d[1] === 0 && +m3d[5] === 0 && +m3d[9] === 0) || // Y Axis
+    (+m3d[2] === 0 && +m3d[6] === 0 && +m3d[10] === 0)) { // Z Axis
```

Comment:
Can you pull this out into separate functions that explain what they mean? It's hard to read what this if statement is checking exactly just reading all the matrix values together.

For example, something like

```js
const xAxisIs0 = +m3d[0] === 0 && +m3d[4] === 0 && +m3d[8] === 0


// later

if (xAxisIs0 || ...)
```